File: baseutils/src/main/java/com/ec/www/view/swiperecycler/utils/ArrayUtils.java
package com.ec.www.view.swiperecycler.utils;

import java.util.Collections;
import java.util.List;

/**
 * Created by huang on 2017/1/20.
 */

public class ArrayUtils {

    public static void swap(List<?> list, int fromPosition, int toPosition) {
        if (fromPosition < toPosition) {
            for (int i = fromPosition; i < toPosition; i++) {
                Collections.swap(list, i, i + 1);
            }
        } else {
            for (int i = fromPosition; i > toPosition; i--) {
                Collections.swap(list, i, i - 1);
            }
        }
    }

}

File: baseutils/src/main/java/com/ec/www/view/swiperecycler/utils/ArrayUtils.kt
package com.ec.www.view.swiperecycler.utils

import java.util.Collections

/**
 * Created by huang on 2017/1/20.
 */

object ArrayUtils {

    fun swap(list: List<*>, fromPosition: Int, toPosition: Int) {
        if (fromPosition < toPosition) {
            for (i in fromPosition..toPosition - 1) {
                Collections.swap(list, i, i + 1)
            }
        } else {
            for (i in fromPosition downTo toPosition + 1) {
                Collections.swap(list, i, i - 1)
            }
        }
    }

}
